apps/emqx/src/emqx_channel.erl

```diff
@@ -276,7 +276,9 @@ init(
     ),
     {NClientInfo, NConnInfo} = take_ws_cookie(ClientInfo, ConnInfo),
     #channel{
-        conninfo = NConnInfo,
+        %% We remove the peercert because it duplicates to what's stored in the socket,
+        %% Saving a copy here causes unnecessary wast of memory (about 1KB per connection).
+        conninfo = maps:put(peercert, undefined, NConnInfo),
```

Comment:
👍

Reply:
just curious, why not delete it?

Reply:
Because the `emqx_types.erl` defines that the `peercert` is a required field of `conninfo()`.

Reply:
How can hooks get access to the peer cert now that it has been removed?

Reply:
Hi @MikeDombo 

it will be fixed in this PR: https://github.com/emqx/emqx/pull/10715
the removal is delayed until `connected` hook is called.